core/src/main/java/org/jruby/RubyKernel.java

```diff
@@ -873,6 +873,42 @@ public static IRubyObject sprintf(ThreadContext context, IRubyObject recv, IRuby
     public static IRubyObject sprintf(IRubyObject recv, IRubyObject[] args) {
         return sprintf(recv.getRuntime().getCurrentContext(), recv, args);
     }
+    public static IRubyObject raise(ThreadContext context, IRubyObject self, IRubyObject arg0) {
+        final Ruby runtime = context.runtime;
+        int argc = 1;
+        boolean forceCause = false;
+
+        // semi extract_raise_opts :
+        IRubyObject cause;
```

Comment:
These are not referenced until line 906, they could be moved down.

Also it looks like argc is assigned `1`, doesn't change, and is then compared with `0`

Reply:
Good catch, I'll look into it.

Reply:
Addressed in b61cf077d3. This `raise` is a 1-arg convenience method, and I just failed to remove all unused logic when I made it.